### PR TITLE
ManageWikiExtensions: Log error if ext is not set in config && allow removing

### PR DESCRIPTION
**Pull request by paladox**
* Logs error if ext does not exist in config.
* Allows removing ext when it does not exist in config.

Bug: T12868

### CODE REVIEW

**Comment by paladox** [comment]
@Universal-Omega could you review/merge please?

File: includes/Helpers/ManageWikiExtensions.php [hunk 1]
```diff
@@ -55,9 +55,17 @@ public function __construct( string $wiki ) {
 			]
 		)->s_extensions ?? '[]';
 
+		$logger = LoggerFactory::getInstance( 'ManageWiki' );
```
Comment (Universal-Omega) [inline includes/Helpers/ManageWikiExtensions.php:58]: Needs `use MediaWiki\Logger\LoggerFactory;`

File: includes/Helpers/ManageWikiExtensions.php [hunk 2]
```diff
@@ -90,16 +98,18 @@ public function add( $extensions ) {
 	/**
 	 * Removes an extension from the 'enabled' list
 	 * @param string|string[] $extensions Either an array or string of extensions to disable
+	 * @param bool $forceRemove Force removing extension incase it is removed from config
 	 */
-	public function remove( $extensions ) {
+	public function remove( $extensions, $forceRemove ) {
```
Comment (Universal-Omega) [inline includes/Helpers/ManageWikiExtensions.php:103]: pass false to forceRemove in overwiteAll below or this will error

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

File: includes/Helpers/ManageWikiExtensions.php [hunk 3]
```diff
@@ -90,16 +98,18 @@ public function add( $extensions ) {
...
 		// We allow remove either one extension (string) or many (array)
 		// We will handle all processing in final stages
 		foreach ( (array)$extensions as $ext ) {
-			if ( !isset( $this->liveExts[$ext] ) ) {
+			if ( !isset( $this->liveExts[$ext] ) && !$forceRemove ) {
 				continue;
 			}
 
-			$this->removedExts[$ext] = $this->liveExts[$ext];
+			$this->removedExts[$ext] = isset( $this->liveExts[$ext] ) ?
```
Comment (Universal-Omega) [inline includes/Helpers/ManageWikiExtensions.php:111]: `$this->removedExts[$ext] = $this->liveExts[$ext] ?? []`

File: includes/Helpers/ManageWikiExtensions.php (hunk 1 above)
Comment (paladox) [reply includes/Helpers/ManageWikiExtensions.php:59]: Thanks, done.

File: includes/Helpers/ManageWikiExtensions.php (hunk 2 above)
Comment (paladox) [reply includes/Helpers/ManageWikiExtensions.php]: I supplied a default value of false to it.

File: includes/Helpers/ManageWikiExtensions.php (hunk 3 above)
Comment (paladox) [reply includes/Helpers/ManageWikiExtensions.php]: Thanks, done.

**Review by Universal-Omega — APPROVED** [review]

**Pull request merged**